framework/bundles/org.jboss.tools.rsp.logging/src/org/jboss/tools/rsp/logging/internal/SLF4JLogReader.java

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.rsp.logging.internal;
+
+import org.osgi.service.log.LogEntry;
+import org.osgi.service.log.LogListener;
+import org.osgi.service.log.LogService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An SLF4J-based log reader.
+ */
+
+public final class SLF4JLogReader implements LogListener {
+	private static final Logger LOG = LoggerFactory.getLogger(SLF4JLogReader.class);
+
+	@Override
+	public void logged(final LogEntry entry) {
+		final int level = entry.getLevel();
```

Comment:
#getLevel is deprecated as are all the constants. The corrected code would look as follows:
```
		final LogLevel level = entry.getLogLevel();
		switch (level) {
		case DEBUG:
			SLF4JLogReader.debug(entry);
			break;
		case ERROR:
			SLF4JLogReader.error(entry);
			break;
		case INFO:
			SLF4JLogReader.info(entry);
			break;
		case WARN:
			SLF4JLogReader.warn(entry);
			break;
		default:
			SLF4JLogReader.warn(entry);
			break;
		}
```